train.py

```diff
@@ -72,8 +72,13 @@ def train(hyp, opt, device, tb_writer=None):
         print('Transferred %g/%g items from %s' % (len(state_dict), len(model.state_dict()), weights))  # report
     else:
         model = Model(opt.cfg, ch=3, nc=nc).to(device)  # create
-
+    
     # Freeze
+    if opt.freeze !='':
+        freeze=[]
+        [freeze.append(x)for x in opt.freeze.split()]
+    else:
+        freeze = ['', ]
     freeze = ['', ]  # parameter names to freeze (full or partial)
```

Comment:
Wouldn't this line override previous array if you go with your current code?

Reply:
@NanoCode012  With reference to Fixes #679,  I added `parser.add_argument('--freeze', type=str, default='', help='freezing gradients')` . If some one add additional argument --freeze with python3 train.py they can enter the parameters which they want to freeze. So default is empty string due to which freeze would have `['', ]` as stated in Fixes #679 else there would be array of the elements to be freezed separated by space bar.

Reply:
I'm sorry if I'm not clear here also, Git review is making a bit hard for me to mention a single line.
https://github.com/ultralytics/yolov5/blob/d2b00dbe38b44a6f80b2adc581b1561cc4c1e56e/train.py#L82

This line would reassigns the value of `freeze`

Reply:
Sorry, I forgot to remove this line. I will update it.

Reply:
@NanoCode012 Updated